packages/react-core/src/components/CalendarMonth/CalendarMonth.tsx

```diff
@@ -280,13 +280,12 @@ export const CalendarMonth = ({
                 const isSelected = isValidDate(selectedDate) && isSameDate(date, selectedDate);
                 const isFocused = isSameDate(date, focusedDate);
                 const isAdjacentMonth = date.getMonth() !== focusedDate.getMonth();
+                const isRangeStart = isValidDate(rangeStart) && isSameDate(date, rangeStart);
                 let isInRange = false;
-                let isRangeStart = false;
                 let isRangeEnd = false;
                 if (isValidDate(rangeStart) && isHoveredDateValid) {
                   if (hoveredDate > rangeStart || isSameDate(hoveredDate, rangeStart)) {
                     isInRange = date > rangeStart && date < hoveredDate;
```

Comment:
If i change the logic in this if statement to:
```
if (hoveredDate > rangeStart || isSameDate(hoveredDate, rangeStart)) {
    isInRange = date > rangeStart && date < selectedDate;
    isRangeEnd = isSameDate(date, selectedDate);
}
```
then the hover behavior updates to look like this:
![Mar-25-2022 08-56-48](https://user-images.githubusercontent.com/26305082/160125016-aaa2d763-1c60-4edf-818c-5df3870c87a2.gif)

I personally think this is more expected? What do you all think @wise-king-sullyman, @thatblindgeye, @mcarrano, @mmenestr ?

Reply:
I do think that's more in line with what I would expect as well.